packages/@react-aria/color/docs/useColorArea.mdx

```diff
@@ -0,0 +1,616 @@
+You can change the default value using the `defaultValue` prop, and the color area will use the color space of the provided value.
+If no `xChannel` or `yChannel` is provided, for the RGB color space, the `red` color channel maps to the horizontal axis or `xChannel`,
+and the `green` color channel maps to the vertical axis or `yChannel`. Similarly, for the HSL and HSB color spaces, the `hue` color
+channel maps to the horizontal axis or `xChannel`, and the `saturation` color channel maps to the vertical axis or `yChannel`.
+
+```tsx example
+<label id="hsb-label-id">x: Hue, y: Saturation</label>
+<ColorArea
+  aria-labelledby="hsb-label-id"
+  defaultValue={'hsb(219, 58%, 93%)'} />
+```
+
+### Controlled
+
+A color area can be made controlled using the `value` prop. The <TypeLink links={statelyDocs.links} type={statelyDocs.exports.parseColor} />
+function is used to parse the initial color from an RGB, HSL or HSB string, stored in state. The `onChange` prop
+is used to update the value in the state when the user drags the thumb. This is the more common usage because it allows to adjust the third
+color channel using a separate control, like a color slider using the [useColorSlider](useColorSlider.html) hook or a color wheel using the
+[useColorWheel](useColorWheel.html) hook.
+
+```tsx example
+import {parseColor} from '@react-stately/color';
+
+function Example() {
+  let [
+    color,
+    setColor
+  ] = React.useState(parseColor('hsb(219, 58%, 93%)'));
+  let [
+    xChannel,
+    yChannel,
+    zChannel
+  ] = color.getColorChannels();
+  return (
+    <>
+      <label id="hsb-label-id-1">x: {color.getChannelName(xChannel, 'en-US')}, y: {color.getChannelName(yChannel, 'en-US')}</label>
+      <div style={{display: 'flex', flexWrap: 'wrap'}}>
+        <div style={{marginRight: '2rem'}}>
+          <ColorArea
+            aria-labelledby="hsb-label-id-1"
+            value={color}
+            onChange={setColor}
+            xChannel={xChannel}
+            yChannel={yChannel} />
+        </div>
+        <div
+          role="img"
+          aria-label={`color swatch: ${color.toString('hsl')}`}
+          style={{
+            backgroundColor: color.toString('css'),
+            forcedColorAdjust: 'none',
+            height: 96,
+            marginRight: '2rem',
+            width: 96
+          }}></div>
+      </div>
+      <label>
+        {color.getChannelName(zChannel, 'en-US')}<br/>
+        <input
+          type="range"
+          min={color.getChannelRange(zChannel).minValue}
+          max={color.getChannelRange(zChannel).maxValue}
+          value={color.getChannelValue(zChannel)}
+          onChange={e => setColor(color.withChannelValue(zChannel, e.target.value))}
+          style={{
+            width: 192
+          }} />
+      </label>
+      <p>Current color value: {color.toString('hsb')}</p>
+    </>
+  );
+}
+```
```

Comment:
I'd like to breakup/simplify this example and the below similar ones a bit more. It's a lot to comprehend beyond the color area itself, which should be the focus of these docs. There are four components in this example: ColorArea itself, a bespoke color swatch, and a ColorSlider (which we could reuse from the other docs page).

I think we should perhaps keep most of the usage examples focused on ColorArea itself, and maybe have a separate section about creating a color picker combining these components. Or maybe that should be on a separate page somewhere since really it could apply on any of the color component docs. Not sure... I do know that we've learned the examples need to be as simple and concise as possible or people's eyes glaze over though. 😄

Reply:
I'll see if I can simplify by reusing the ColorSlider from the useColorSlider docs. The color swatch image to show the current value was added to satisfy @snowystinger's request that we better demonstrate the difference between `onChange` and `onChangeEnd`: https://github.com/adobe/react-spectrum/pull/3268#discussion_r912296596

Reply:
One thing I'm also trying to demonstrate is how to use xChannel, yChannel with an external control for the zChannel, which is kind of essential to understanding the ColorArea.

Reply:
That makes sense. We could perhaps combine the examples for onChange and onChangeEnd, which are essentially the same code-wise. We could do something more like what is in the Spectrum example: https://reactspectrum.blob.core.windows.net/reactspectrum/eaa1dccfca778a634015b5de2fdc7b1f0f2b2447/docs/react-spectrum/ColorArea.html#events, where both values are shown.